File: JavaPractice/LinkedList.java
// Linked List

public class LinkedList{
	Node head;
	static class Node{
		int data;
		Node next;

		//constructor
		Node(int d){
			data = d;
			next = null;
		}
	}

	public static LinkedList insert(LinkedList list, int data)
	{
		Node new_node = new Node(data);
		new_node.next = null;

		if(list.head == null)
		{
			list.head = new_node;
		}
		else{
			Node last = list.head;
			while(last.next!=null){
				last = last.next;
			}

			last.next = new_node;
		}

		return list;
	}

	public static LinkedList insertIndex(LinkedList list, int index, int data)
	{
		Node curr_node = list.head;
		Node new_node = new Node(data);
		for(int i=0;i<index-1;i++){
			curr_node = curr_node.next;
			if(curr_node==null){
				System.out.print("Can't insert element");
				return list;
			}
		}
		new_node.next = curr_node.next;
		curr_node.next = new_node;
		return list;
	}

	public static LinkedList deleteNode(LinkedList list, int index)
	{
		Node curr_node = list.head, prev_node = null;
		if(index==0){
			curr_node = curr_node.next;
			list.head = curr_node;
			return list;
		}
		for(int i=0;i<index;i++){
			if(curr_node==null){
				System.out.print("Nothing to delete");
				return list;
			}
			else{
				prev_node = curr_node;
				curr_node = curr_node.next;

			}
		}
		prev_node.next = curr_node.next;
		return list;
	}

	public static void printList(LinkedList list)
	{
		Node curr_node = list.head;
		System.out.println("--------Linked List---------");
		while(curr_node!= null){
			System.out.println(curr_node.data);
			curr_node = curr_node.next;
		}
	}

	public static void main(String[] args) {
		LinkedList list = new LinkedList();

		list = insert(list, 23);
		list = insert(list, 34);
		list = insert(list, 54);
		list = insert(list, 12);
		list = insert(list, 45);
		list = insert(list, 78);
		list = insert(list, 1);
		list = insert(list, 11);

		printList(list);
		list = insertIndex(list, 3, 1111);
		printList(list);
		deleteNode(list, 3);
		printList(list);
	}
}

File: ScratchAlgo/linkedList.cpp
#include<bits/stdc++.h>
using namespace std;

class Node
{
public:
	int data;
	Node* next;

	Node(int x)
	{
		data = x;
		next = NULL;
	}
};


class LinkedList
{
	Node* head;
public:
	LinkedList()
	{
		head = NULL;
	}

	void insertNode(int);

	void printList();

	void deleteNode(int);
};

void LinkedList::insertNode(int data)
{
	Node* newNode = new Node(data);
	if(head == NULL)
	{
		head = newNode;
		return;
	}

	Node* temp = head;
	while(temp->next)
	{
		temp = temp->next;
	}
	temp->next = newNode;
}

void LinkedList::printList()
{
	Node* temp = head;
	if(head == NULL)
	{
		cout<<"Empty list"<<endl;
		return;
	}

	while(temp)
	{
		cout<<temp->data<<" ";
		temp = temp->next;
	}
	cout<<endl;
}

void LinkedList::deleteNode(int dnode)
{
	Node* temp1 = head;
	Node* temp2 = NULL;

	if(!head)
	{
		cout<<"List is Empty"<<endl;
		return;
	}

	while(temp1 && temp1->data != dnode)
	{
		temp2 = temp1;
		temp1 = temp1->next;
	}

	temp2->next = temp1->next;
}

int main()
{
	LinkedList list;
	list.insertNode(1);
	list.insertNode(10);
	list.insertNode(2);
	list.insertNode(3);
	list.insertNode(4);
	list.printList();

	list.deleteNode(10);
	list.printList();
	return 0;
}